src/complex/Core/Parameters/utils/FilePathGenerator.hpp

```diff
@@ -0,0 +1,188 @@
+  {
+    return fileList;
+  }
+  int32_t index = 0;
+  // Create the actual format string
+  std::stringstream fmt_string_stream;
+  fmt_string_stream << "{}/{}{:0" << info.getPaddingDigits() << "d}{}{}";
+
+  bool missingFiles = false;
+  for(int32_t i = 0; i < (info.getEndIndex() - info.getStartIndex()) + 1; i = i + info.getIncrementIndex())
+  {
+    if(info.getOrdering() == 0)
+    {
+      index = info.getStartIndex() + i;
+    }
+    else
+    {
+      index = info.getEndIndex() - i;
+    }
+
+    std::string filePath = fmt::format(fmt_string_stream.str(), info.getInputPath(), info.getFilePrefix(), index, info.getFileSuffix(), info.getFileExtension());
+
+    if(validatePaths && !fs::exists(filePath))
+    {
+      missingFiles = true;
+    }
+
+    fileList.push_back(filePath);
+  }
+
+  if(missingFiles)
+  {
+    return {};
+  }
+  return fileList;
+}
+
+/**
+ * @brief generateFileList This method will generate a file list in the correct order of the files that should
+ * be imported int32_to an h5ebsd file
+ * @param start Z Slice Start
+ * @param end S Slice End
+ * @param increment How much to increment each item
+ * @param hasMissingFiles Are any files missing
+ * @param stackLowToHigh Is the Slice order low to high
+ * @param filename Example File Name
+ * @return
+ */
+std::vector<std::string> GenerateFileList(int32_t start, int32_t end, int32_t increment, bool& hasMissingFiles, bool stackLowToHigh, const std::string& inputPath, const std::string& filePrefix,
+                                          const std::string& fileSuffix, const std::string& fileExtension, int32_t paddingDigits)
+{
+  std::vector<std::string> fileList;
+
+  if(!fs::exists(inputPath))
+  {
+    return fileList;
+  }
+  int32_t index = 0;
+
+  std::stringstream fmt_string_stream;
+
+  fmt_string_stream << "{}/{}{:0" << paddingDigits << "d}{}{}";
+
+  bool missingFiles = false;
+  for(int32_t i = 0; i < (end - start) + 1; i = i + increment)
+  {
+    if(stackLowToHigh)
+    {
+      index = start + i;
+    }
+    else
+    {
+      index = end - i;
+    }
+
+    std::string filePath = fmt::format(fmt_string_stream.str(), inputPath, filePrefix, index, fileSuffix, fileExtension);
+
+    if(!fs::exists(filePath))
+    {
+      missingFiles = true;
+    }
+
+    fileList.push_back(filePath);
+  }
+
+  hasMissingFiles = missingFiles;
+
+  return fileList;
+}
+
+/**
+ * @brief generateFileList This method will generate a file list in the correct order of the files that should
+ * be imported int32_to an h5ebsd file
+ * @param start Z Slice Start
+ * @param end S Slice End
+ * @param hasMissingFiles Are any files missing
+ * @param stackLowToHigh Is the Slice order low to high
+ * @param filename Example File Name
+ * @return
+ */
+std::vector<std::string> GenerateVectorFileList(int32_t start, int32_t end, int32_t compStart, int32_t compEnd, bool& hasMissingFiles, bool stackLowToHigh, const std::string& inputPath,
```

Comment:
Are the other functions in the file ready?

Reply:
Removed dead code